Tests/Parser/JmsMetadataParserTest.php

```diff
@@ -582,6 +487,115 @@ public function testParserWithInline()
         );
     }
 
+    public function testParserWithDiscriminator()
+    {
+        $metadataFactory     = $this->createMock(MetadataFactoryInterface::class);
+        $docCommentExtractor = $this->getMockBuilder('Nelmio\ApiDocBundle\Util\DocCommentExtractor')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $inputMainClass = 'Nelmio\ApiDocBundle\Tests\Fixtures\Model\JmsWithDiscriminators';
+        $inputDiscriminatorClass = 'Nelmio\ApiDocBundle\Tests\Fixtures\Model\DiscriminatorClass';
+        $discriminatorFieldName = 'type';
+
+        $propertyMetadataFoo       = new PropertyMetadata($inputMainClass, 'foo');
+        $propertyMetadataFoo->type = array(
+            'name' => 'string',
+        );
+
+        $propertyMetadataBar        = new PropertyMetadata($inputDiscriminatorClass, 'bar');
+        $propertyMetadataBar->type  = array(
+            'name' => 'string',
+        );
+
+        $metadataMainClass = new ClassMetadata($inputMainClass);
+        $metadataMainClass->addPropertyMetadata($propertyMetadataFoo);
+        $metadataMainClass->setDiscriminator($discriminatorFieldName,
+            array(
+                'TYPE_1' => $inputDiscriminatorClass,
+            )
+        );
+
+        $metadataDiscriminatorClass = new ClassMetadata($inputDiscriminatorClass);
+        $metadataDiscriminatorClass->addPropertyMetadata($propertyMetadataFoo);
+        $metadataDiscriminatorClass->addPropertyMetadata($propertyMetadataBar);
+
+        $metadataFactory->expects($this->any())
+            ->method('getMetadataForClass')
+            ->will($this->returnValueMap(
+                array(
+                    array($inputMainClass, $metadataMainClass),
+                    array($inputDiscriminatorClass, $metadataDiscriminatorClass)
+                )
+            ));
+
+        $propertyNamingStrategy = new CamelCaseNamingStrategy();
+
+        $jmsMetadataParser = new JmsMetadataParser($metadataFactory, $propertyNamingStrategy, $docCommentExtractor);
+
+        // No group specified.
+        $output = $jmsMetadataParser->parse(
+            array(
+                'class'   => $inputMainClass,
+                'groups'  => array(),
+            )
+        );
+
+        $this->assertEquals(
+            array(
```

Comment:
The array could be in static method or property. The test would be more readable.

Reply:
I tried to keep it as close to original as possible, even at the expense of readability. This way it's easier to compare with https://github.com/nelmio/NelmioApiDocBundle/tree/2.13.3